sdk/storage/Azure.Storage.Blobs/tests/VersionLevelWormTests.cs

```diff
@@ -0,0 +1,890 @@
+                await operation.WaitForCompletionAsync();
+            }
+
+            // Assert
+            Response<BlobProperties> propertiesResponse = await destBlob.GetPropertiesAsync();
+            Assert.AreEqual(expectedImmutabilityPolicyExpiry, propertiesResponse.Value.ImmutabilityPolicyExpiresOn);
+            Assert.AreEqual(immutabilityPolicy.PolicyMode.ToString().ToLowerInvariant(), propertiesResponse.Value.ImmutabilityPolicyMode);
+            Assert.IsTrue(propertiesResponse.Value.HasLegalHold);
+        }
+
+        private async Task <DisposingVersionLevelWormContainer> GetTestVersionLevelWormContainer(TenantConfiguration tenantConfiguration)
+        {
+            StorageSharedKeyCredential sharedKeyCredential = new StorageSharedKeyCredential(TestConfigOAuth.AccountName, TestConfigOAuth.AccountKey);
+            Uri serviceUri = new Uri(TestConfigOAuth.BlobServiceEndpoint);
+            BlobServiceClient blobServiceClient = InstrumentClient(new BlobServiceClient(serviceUri, sharedKeyCredential, GetOptions()));
+            BlobContainerClient containerClient = InstrumentClient(blobServiceClient.GetBlobContainerClient(GetNewContainerName()));
+
+            DisposingVersionLevelWormContainer disposingVersionLevelWormContainer = new DisposingVersionLevelWormContainer(
+                tenantConfiguration,
+                containerClient,
+                Mode);
+            await disposingVersionLevelWormContainer.CreateAsync();
+            return disposingVersionLevelWormContainer;
+        }
+
+        private DateTimeOffset RoundToNearestSecond(DateTimeOffset initalDateTimeOffset)
+            => new DateTimeOffset(
+                year: initalDateTimeOffset.Year,
+                month: initalDateTimeOffset.Month,
+                day: initalDateTimeOffset.Day,
+                hour: initalDateTimeOffset.Hour,
+                minute: initalDateTimeOffset.Minute,
+                second: initalDateTimeOffset.Second,
+                offset: TimeSpan.Zero);
+    }
+
+#pragma warning disable SA1402 // File may only contain a single type
+    public class DisposingVersionLevelWormContainer : IAsyncDisposable
+#pragma warning restore SA1402 // File may only contain a single type
+    {
+        public BlobContainerClient Container;
+
+        private TenantConfiguration _tenantConfiguration;
+        private StorageManagementClient _storageManagementClient;
+
+        private RecordedTestMode _testMode;
+
+        public DisposingVersionLevelWormContainer(
+            TenantConfiguration tenantConfiguration,
+            BlobContainerClient containerClient,
+            RecordedTestMode recordedTestMode)
+        {
+            _tenantConfiguration = tenantConfiguration;
+            Container = containerClient;
+            _testMode = recordedTestMode;
+        }
+
+        public async Task CreateAsync()
+        {
+            if (_testMode == RecordedTestMode.Playback)
+            {
+                return;
```

Comment:
I know this is cheesy, but it saved us a ton of time.

We don't need to actually be testing the Management Plane SDK here, we just need to to setup and tear down our tests in `Live` and `Recorded` modes.